src/components/form/form.RadioGroup.tsx

```diff
@@ -0,0 +1,41 @@
+import { Form } from "./form";
+import { useForm } from 'react-hook-form';
+
+type FormData = {
+    diabetesType: string;
+};
+
+const DiabetesForm = () => {
+    const {register, handleSubmit, formState: { errors }} = useForm<FormData>();
+
+    const onSubmit = (data: FormData) => {
+        if (!errors.diabetesType) {
+        alert(`You selected ${data.diabetesType}`)
+        } else {
+            alert( 'Please select a diabetes type');
+        }
+    };
+
+    return (
+        <Form onSubmit={handleSubmit(onSubmit)}>
+            <p>Choose your diabetes type</p>
+            <div {...register('diabetesType', {required: true})}>
+                <label>
+                    <input type= "radio" value= "Type 1"/>
+                    Type 1
```

Comment:
I would pass a prop here to the inputs that allow for it to be more reusable for other devs. Example ``` <label>
                    <input type= "radio" value= {value}/>
                    {option1}
                </label>```  This will allow this component to be used for more forms than just selecting types.